Automattic/jetpack
# Social: Clean up the creating connection spinner for confirm button

Currently, we close the modal immediately after a user clicks confirm on the connection confirmation screen. But, if they click on connect button again, the confirmation screen will show loading indicator for confirm button. This is not ideal.

This PR is a follow-up of this comment https://github.com/Automattic/jetpack/pull/37730#pullrequestreview-2101684523

Fixing that means users can add connections in parallel.

## Proposed changes:

* Do not disable the confirm button when connection creation is in progress
* Fix the optimistic account display name when selecting not the first account when there is more than one account to connect.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:







- Ensure that feature flag is ON - `define( 'JETPACK_SOCIAL_USE_ADMIN_UI_V1', true );`
- Test the following for both Jetpack and Social plugin and the editor.
    - Goto connections management UI
    - Add a connection and click on "Confirm"
    - Immediately open the connection modal again and connect another account
    - To simulate a time taking request, add add `sleep( 10 );` as the first statement inside `create_publicize_connection` method in `projects/packages/publicize/src/class-rest-controller.php`
    - Confirm that while previous connection creation request is in progress, the "Confirm" button is not disabled
    - Click on the "Confirm" button
    - Confirm that the connection is created successfully
    - To verify the display name fix when there are more than one account to connect, select the second account on the confirmation screen
    - Confirm that the display name is correct and not of the first account in the list on confirmation page.
   

https://github.com/Automattic/jetpack/assets/18226415/12dfcfa6-5f59-42dd-b01c-19e8e5380d7e
